includes/platform-checkout/class-platform-checkout-utilities.php

```diff
@@ -25,8 +25,21 @@ class Platform_Checkout_Utilities {
 	public function should_enable_platform_checkout( $gateway ) {
 		$is_platform_checkout_eligible = WC_Payments_Features::is_platform_checkout_eligible(); // Feature flag.
 		$is_platform_checkout_enabled  = 'yes' === $gateway->get_option( 'platform_checkout', 'no' );
-		$disable_for_subscription      = ! WC_Payments_Features::is_platform_checkout_subscriptions_enabled() && $this->is_subscription_item_in_cart();
+		$has_subscription_in_cart      = $this->is_subscription_item_in_cart();
+		$disable_for_subscription      = ! WC_Payments_Features::is_platform_checkout_subscriptions_enabled() && $has_subscription_in_cart;
 
-		return $is_platform_checkout_eligible && $is_platform_checkout_enabled && ! $disable_for_subscription;
+		$enable_platform_checkout = $is_platform_checkout_eligible && $is_platform_checkout_enabled && ! $disable_for_subscription;
+
+		if ( $enable_platform_checkout && $has_subscription_in_cart && ! is_user_logged_in() ) {
+			wp_localize_script(
```

Comment:
I wouldn't expect a method for calculating whether the platform checkout should be enabled to call `wp_localize_script()`. I also don't think we want to tie this side-effect to checking whether the platform checkout should be enabled. 

So, I'd say to move this localization to a context where it fits better. Maybe in [class-wc-payment-gateway-wcpay.php](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php) where we're handling scripts? Just throwing a quick idea for the location.

Reply:
Makes sense, fixed on https://github.com/Automattic/woocommerce-payments/pull/4436/commits/bdcd132af496beb9d115b8594e658b9fc5d3b822.